[docs] create advanced guides directory; rename files 1/N

angelinalg opened the pull request:
Implement new directory structure per plan. This is one of many to come. I wanted to start with just the directory structure changes and get that merged asap to head off merge conflicts.

@edoakes, @akshay-anyscale and I discussed removing the User Guides directory. With the restructuring, the number of non-Advanced guides was only 5-6 guides and we thought they would be more discoverable at a higher level. Take a look and let us know what you think, please.

## Why are these changes needed?

cc: @akshay-anyscale 

## Related issue number

<!-- For example: "Closes #1234" -->

## Checks

- [x ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

akshay-anyscale commented:
Taking another look at the content, I believe Add End to end Fault tolerance should go under the production guide (it discusses a lot of K8s concepts and is important for customers deploying to production). The in-place updates should be moved up in the advance guides (based on https://docs.google.com/document/d/1vX25180F2VkwsWaQR9Yh4fyUDus-7pVmTHQXpfD1n3Q/edit)

angelinalg commented:
Ah! Sorry, I misplaced the a `In-place Updates`. No problem to move the `End-to-End` back into the Production Guide. I should move it back to after `Deploying on K8s`, right?

edoakes commented on doc/source/_toc.yml:
```diff
@@ -256,27 +256,33 @@ parts:
+          - file: serve/advanced-guides/index
+            sections:
+              - file: serve/advanced-guides/app-builder-guide
+              - file: serve/advanced-guides/http-guide
```
I don't think this one should be advanced, it has some pretty basic info like how to use FastAPI integration

edoakes commented:
@angelinalg the structure sounds good, agree with making them more discoverable. Only comment is I don't think `http-guide` should be nested under "advanced" because it's pretty basic/important info.

akshay-anyscale commented:
LGTM, agree with Ed's comment on moving up the FastAPI/HTTP guides

angelinalg commented:
@edoakes, ready for your review. I moved the FastAPI doc up to the Production Guide. I also fixed the order of appearance in the side nav for the In-place updates to Serve. If you approve, please merge. Thanks!

edoakes commented:
@angelinalg looks like there are relevant failures:
https://buildkite.com/ray-project/oss-ci-build-pr/builds/25381#0188b67a-fc99-4cf9-a390-30f2351f9caa/1309-1908

angelinalg commented:
@edoakes, I think I cleaned up all of the related messages. Would you mind taking another look?
